packages/back-end/src/scim/users/createUser.ts

```diff
@@ -0,0 +1,75 @@
+import { Response } from "express";
+import { cloneDeep } from "lodash";
+import {
+  addMemberToOrg,
+  convertMemberToManagedByIdp,
+  expandOrgMembers,
+} from "../../services/organizations";
+import { OrganizationInterface } from "../../../types/organization";
+import { ScimUserPutOrPostRequest } from "../../../types/scim";
+import {
+  createUser as createNewUser,
+  getUserByEmail,
+} from "../../services/users";
+
+export async function createUser(req: ScimUserPutOrPostRequest, res: Response) {
+  const { externalId, displayName, userName } = req.body;
+
+  const org: OrganizationInterface = req.organization;
+
+  const expandedMembers = await expandOrgMembers(org.members);
+  const existingOrgMember = expandedMembers.find(
+    (member) => member.email === userName
+  );
+
+  const role = org.settings?.defaultRole?.role || "readonly";
+  const responseObj = cloneDeep(req.body);
+
+  try {
+    if (existingOrgMember && existingOrgMember.managedByIdp) {
+      // Return an error if user exists and they're managed by an external IDP
+      return res.status(409).json({
+        schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+        scimType: "uniqueness",
+        detail: "User already exists in this organization",
+        status: 409,
+      });
+    } else if (existingOrgMember && !existingOrgMember.managedByIdp) {
+      // If created through GrowthBook, update the user with the externalId and managedByIdp: true
+      await convertMemberToManagedByIdp({
+        organization: org,
+        userId: existingOrgMember.id,
+        externalId: externalId ? externalId : undefined,
+      });
+
+      responseObj.id = existingOrgMember.id;
+    }
+
+    let user = await getUserByEmail(userName);
+
+    if (!user) {
+      user = await createNewUser(displayName, userName);
+    }
+
+    await addMemberToOrg({
+      organization: org,
+      userId: user.id,
+      role,
+      limitAccessByEnvironment: false,
+      environments: [],
+      projectRoles: [],
+      externalId,
+      managedByIdp: true,
+    });
+
+    responseObj.id = user.id;
```

Comment:
Shouldn't all of this be in an `else` block?  If there's already an existing member in the org, we don't need to do any of this stuff.